examples/nats-bench.go

```diff
@@ -134,22 +104,22 @@ func runPublisher(startwg, donewg *sync.WaitGroup, opts nats.Options, numMsgs in
 	args := flag.Args()
 	subj := args[0]
 	var msg []byte
-	if(messageSize > 0) {
-		msg = make([]byte, messageSize)
+	if msgSize > 0 {
+		msg = make([]byte, msgSize)
 	}
 
 	start := time.Now()
 
 	for i := 0; i < numMsgs; i++ {
 		nc.Publish(subj, msg)
 	}
-	pubStatChan <- newStats(start, time.Now(), nc, true)
+	benchmark.AddPubSample(bench.NewSample(numMsgs, msgSize, start, time.Now(), nc))
```

Comment:
For the publisher, you should move the `nc.Flush()` ahead of the capture of the time here. That will be more representative of all data flushed to the server.